tests/functional/resolvers.ts

```diff
@@ -1023,6 +1029,60 @@ describe("Resolvers", () => {
       expect(fieldResolverMethodResult).toBeLessThanOrEqual(1);
     });
 
+    it("should fail when a query exceeds the max allowed complexity", () => {
+      const query = `query {
+        sampleQuery {
+          complexResolverMethod
+        }
+      }`;
+      const ast = parse(query);
+      const typeInfo = new TypeInfo(schema);
+      const context = new ValidationContext(schema, ast, typeInfo);
+      const visitor = new QueryComplexity(context, {
+        maximumComplexity: 5,
+      });
+      visit(ast, visitWithTypeInfo(typeInfo, visitor));
+      expect(context.getErrors().length).toEqual(1);
+      expect(context.getErrors()[0].message).toEqual(
+        "The query exceeds the maximum complexity of 5. Actual complexity is 11",
+      );
+    });
+
+    it("should succeed when a query does not exceed the max allowed complexity", () => {
+      const query = `query {
+        sampleQuery {
+          complexResolverMethod
+        }
+      }`;
+      const ast = parse(query);
+      const typeInfo = new TypeInfo(schema);
+      const context = new ValidationContext(schema, ast, typeInfo);
+      const visitor = new QueryComplexity(context, {
+        maximumComplexity: 12,
+      });
+      visit(ast, visitWithTypeInfo(typeInfo, visitor));
+      expect(context.getErrors().length).toEqual(0);
+    });
+
+    it("Complexity of a field should be overridden by complexity of a field resolver", () => {
+      const query = `query {
+        sampleQuery {
+          fieldResolverMethod
+        }
+      }`;
+      const ast = parse(query);
```

Comment:
Could you create a helper function in this `describe` suite to avoid duplication in this three `it` test cases?